tests/core/test_serialization.py

```diff
@@ -696,6 +704,133 @@ def test_checkpoint_config_and_nested_objects_are_serialized(
+                },
+                "id": "06871341-f028-4f1f-b8e8-a559ab9f62e1",
+            },
+        ],
+    }
+
+    """CheckpointConfig and nested objects like CheckpointValidationConfig should be serialized appropriately with/without optional params."""
+    observed_dump = checkpointConfigSchema.dump(checkpoint_config)
+    assert observed_dump == expected_serialized_checkpoint_config
+
+    loaded_data = checkpointConfigSchema.load(observed_dump)
+    observed_load = CheckpointConfig(**loaded_data)
+    assert checkpointConfigSchema.dump(observed_load) == checkpointConfigSchema.dump(
+        checkpoint_config
+    )
+
+
+def test_serialization_of_datasource_with_nested_objects_spark(spark_session):
+    conf: Iterable[Tuple[str, str]] = spark_session.sparkContext.getConf().getAll()
+    datasource_config: DatasourceConfig = DatasourceConfig(
+        name="taxi_data",
+        class_name="Datasource",
+        module_name="great_expectations.datasource",
+        execution_engine=ExecutionEngineConfig(
+            class_name="SparkDFExecutionEngine",
+            module_name="great_expectations.execution_engine.sparkdf_execution_engine",
+        ),
+        data_connectors={
+            "configured_asset_connector": DataConnectorConfig(
+                class_name="ConfiguredAssetFilesystemDataConnector",
+                module_name="great_expectations.datasource.data_connector.configured_asset_filesystem_data_connector",
+                assets={
+                    "my_asset": AssetConfig(
+                        class_name="Asset",
+                        module_name="great_expectations.datasource.data_connector.asset",
+                        batch_spec_passthrough={
+                            "reader_options": {"header": True},
+                        },
+                    )
+                },
+            )
+        },
+    )
+    expected_serialized_datasource_config: dict = {
+        "data_connectors": {
+            "configured_asset_connector": {
+                "assets": {
+                    "my_asset": {
+                        "batch_spec_passthrough": {"reader_options": {"header": True}},
+                        "class_name": "Asset",
+                        "module_name": "great_expectations.datasource.data_connector.asset",
+                    }
+                },
+                "class_name": "ConfiguredAssetFilesystemDataConnector",
+                "module_name": "great_expectations.datasource.data_connector.configured_asset_filesystem_data_connector",
+            }
+        },
+        "execution_engine": {
+            "class_name": "SparkDFExecutionEngine",
+            "module_name": "great_expectations.execution_engine.sparkdf_execution_engine",
+        },
+        "module_name": "great_expectations.datasource",
+        "class_name": "Datasource",
+        "name": "taxi_data",
+    }
+
+    observed_dump = datasourceConfigSchema.dump(obj=datasource_config)
```

Comment:
Think we could take all of this and combine into one test `test_serialization_something_about_spark` and parameterize the schema and expected return values.

Reply:
For this one I dont feel as comfortable because we are testing two separate objects `CheckpointConfig` and `DatasourceConfig`. I feel like the rest of the tests keep them separated.

Reply:
As a follow-up to this PR, I'm planning on adding 2 more tests (one as a `CheckpointConfig` test and one as a `DatasourceConfig` test), where we do the `pre_dump()` logic on the `schema` object. I think that would be a more appropriate way to parameterize the values.... what do you think?

Reply:
works for me! thank you 🙇🏽